src/SyGuS.ml

```diff
@@ -111,28 +118,21 @@ let parse_sexps (sexps : Sexp.t list) : t =
                     in if List.mem !variables new_var ~equal:(fun x y -> String.equal (fst x) (fst y))
                        then raise (Parse_Exn ("Multiple declarations of variable " ^ (fst new_var)))
                        else variables := new_var :: !variables
-              | List [ (Atom "declare-fun") ; name ; args ; rtype ]
-                -> if args <> List [] then raise (Parse_Exn "Only nullary function (i.e. variable) declarations supported.") else
-                   let new_var = parse_variable_declaration (List [name ; rtype])
-                    in if List.mem !variables new_var ~equal:(fun x y -> String.equal (fst x) (fst y))
-                       then raise (Parse_Exn ("Multiple declarations of variable " ^ (fst new_var)))
-                       else variables := new_var :: !variables
-              | List ( (Atom "declare-primed-var") :: sexps )
-                -> let _var, _type = parse_variable_declaration (List sexps)
-                    in if List.mem !variables (_var, _type) ~equal:(fun x y -> String.equal (fst x) (fst y))
-                       then raise (Parse_Exn ("Multiple declarations of variable " ^ _var))
-                       else variables := (_var, _type) :: (_var ^ "!", _type) :: !variables
               | List ( (Atom "define-fun") :: func_sexps )
                 -> let (func, fconsts) = parse_define_fun func_sexps
                     in if List.mem !funcs func ~equal:(fun x y -> String.equal x.name y.name)
                        (* FIXME: SyGuS format allows overloaded functions with different signatures *)
                        then raise (Parse_Exn ("Multiple definitions of function " ^ func.name))
-                       else funcs := func :: !funcs ; consts := fconsts @ !consts
+                       else funcs := func :: !funcs ; consts := fconsts @ !consts ;
+                       (if String.equal func.name "trans_fun"
+                        then variables := func.args)
               | List [ (Atom "inv-constraint") ; (Atom _invf_name) ; (Atom _pref_name)
                                                ; (Atom _transf_name) ; (Atom _postf_name) ]
                 -> pref_name := _pref_name ; transf_name := _transf_name ; postf_name := _postf_name
-                 ; if not (String.equal !invf_name _invf_name)
-                   then raise (Parse_Exn ("Invariant function [" ^ _invf_name ^ "] not declared"))
+                 ; (if not (String.equal !invf_name _invf_name)
+                   then raise (Parse_Exn ("Invariant function [" ^ _invf_name ^ "] not declared")))
```

Comment:
It would be a good idea to also do similar checks (like the above 2 lines) for pre, trans and post function names.